tfjs-backend-webgpu/src/kernels/matmul_packed_vec4_webgpu.ts

```diff
@@ -308,4 +451,82 @@ export class MatMulPackedVec4Program implements WebGPUProgram {
     `;
     return userCode;
   }
+
+  getUserCodeWgsl(): string {
+    const sampleA = this.fitA ?
+        `return A.numbers[batch * batchASize + row * uniforms.dimInner / 4u + col]` :
+        `if (coordsInBounds2D(vec2<u32>(row, col * 4u), vec2<u32>(uniforms.dimAOuter, uniforms.dimInner))) {
+            return A.numbers[batch * batchASize + row * uniforms.dimInner / 4u + col];
+        }
+        return vec4<f32>(0.0)`;
+
+    const sampleB = this.fitB ?
+        `return B.numbers[batch * batchBSize + row * uniforms.dimBOuter / 4u + col]` :
+        `if(coordsInBounds2D(vec2<u32>(row, col * 4u), vec2<u32>(uniforms.dimInner, uniforms.dimBOuter))) {
+             return B.numbers[batch * batchBSize + row * uniforms.dimBOuter / 4u + col];
+        }
+        return vec4<f32>(0.0)`;
+
+    let activationSnippet = '', applyActivationSnippet = '';
+    if (this.activation) {
+      const activationOp = mapActivationToShaderProgram(
+          this.activation, this.isVec4, this.useWgsl);
+      if (this.hasPreluActivationWeights) {
+        activationSnippet =
+            `fn activation(a : vec4<f32>, outCoord : vec3<u32>) -> vec4<f32> {
+                  let b = getPreluActivationWeightsAtOutCoordsByCoords(outCoord);
+                  ${activationOp}
+                }`;
+      } else {
+        activationSnippet = `
+            fn activation(a : vec4<f32>, outCoord : vec3<u32>) -> vec4<f32> {
+              ${activationOp}
+            }`;
+      }
+
+      applyActivationSnippet = 'value = activation(value, outCoord);';
+    }
+    const addBiasSnippet = this.addBias ?
+        'value = value + getBiasAtOutCoordsByCoords(outCoord);' :
+        '';
+
+    const varSnippet = `let dimInner = uniforms.aShape[2];`;
```

Comment:
Why can't we use `var dimInner = uniforms.dimInner;` ?

Reply:
done